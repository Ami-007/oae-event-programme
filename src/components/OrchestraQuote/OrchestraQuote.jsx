import React from "react";
import "./OrchestraQuote.scss";

const OrchestraQuote = (props) => {
  const { quote, author } = props;
  return (
    <div className="orchestraquote">
      <q className="orchestraquote__content">{quote}</q>
      <p className="orchestraquote__author">-{author}</p>
    </div>
  );
};

export default OrchestraQuote;
