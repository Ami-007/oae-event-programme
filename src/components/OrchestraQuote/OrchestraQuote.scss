@use "../../assets/sass/typography" as *;
@use "../../assets/sass/variables" as *;

.orchestra-quote {
  background-color: var(--theme-color-primary);
  color: var(--theme-color-text);
  height: 8.37rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  font-family: $modern-content;
  font-size: 1.125rem;

  &__content {
    font-weight: 700;
    line-height: 1.688rem;
    margin-top: 0.313rem;
  }

  &__author {
    font-weight: 400;
    margin-top: 0.313rem;
    margin-bottom: 0.626rem;
  }
}
