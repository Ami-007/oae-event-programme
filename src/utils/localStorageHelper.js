export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
