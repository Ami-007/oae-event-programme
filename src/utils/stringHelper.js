export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
