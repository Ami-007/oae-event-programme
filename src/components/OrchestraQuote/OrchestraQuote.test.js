import OrchestraQuote from "./OrchestraQuote";
import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";

test("Renders OrchestraQuote component on homepage", () => {
  const { container } = customRender(
    <OrchestraQuote
      quote="COMPLIANCE IS NOT HEALTHY. WHAT I WANT IS A DIALOGUE."
      author="Vladmir Jurowski"
    />
  );

  expect(container).toMatchSnapshot();
});

test('Should expect to see the following quote "COMPLIANCE IS NOT HEALTHY. WHAT I WANT IS A DIALOGUE."', () => {
  customRender(
    <OrchestraQuote
      quote="COMPLIANCE IS NOT HEALTHY. WHAT I WANT IS A DIALOGUE."
      author="Vladmir Jurowski"
    />
  );

  const quote = screen.getByText(
    /COMPLIANCE IS NOT HEALTHY. WHAT I WANT IS A DIALOGUE./
  );
  const author = screen.getByText(/Vladmir Jurowski/);

  expect(quote).toBeInTheDocument;
  expect(author).toBeInTheDocument;
});
