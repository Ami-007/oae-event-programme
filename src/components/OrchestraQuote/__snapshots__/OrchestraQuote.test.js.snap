// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders OrchestraQuote component on homepage 1`] = `
<div>
  <div
    class="orchestra-quote"
  >
    <q
      class="orchestra-quote__content"
    >
      COMPLIANCE IS NOT HEALTHY. WHAT I WANT IS A DIALOGUE.
    </q>
    <p
      class="orchestra-quote__author"
    >
      -
      Vladmir Jurowski
    </p>
  </div>
</div>
`;
